packages/plugin-ext/src/main/browser/tasks-main.ts

```diff
@@ -95,6 +96,40 @@ export class TasksMainImpl implements TasksMain {
         this.disposables.set(handle, disposable);
     }
 
+    async $fetchTasks(taskVersion: string | undefined, taskType: string | undefined): Promise<TaskDto[]> {
+        if (taskVersion && !taskVersion.startsWith('2.')) { // Theia does not support 1.x or earlier task versions
+            return [];
+        }
+
+        let found: TaskConfiguration[] = [];
+        const tasks = [...this.taskService.getConfiguredTasks(), ...(await this.taskService.getProvidedTasks())];
+        if (taskType) {
+            found = tasks.filter(t => {
+                if (ContributedTaskConfiguration.is(t)) {
```

Comment:
The reason that I am creating a DRAFT PR is that `ContributedTaskConfiguration.is()` is buggy, as @RomanNikitenko pointed out, we should have a more reliable way to check which tasks are contributed which are not.

Reply:
@elaihau thank you for your support, I'm going to complete my current issue and review the PR

Reply:
If I understand correctly, the PR depends on changes https://github.com/theia-ide/theia/pull/5975 and will be updated after merge https://github.com/theia-ide/theia/pull/5975.
@elaihau Is it correct?